prow/apis/prowjobs/v1/types.go

```diff
@@ -163,13 +163,18 @@ type ProwJobSpec struct {
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
 
 	// RerunPermissions holds information about which users can rerun the job
-	RerunPermissions *RerunPermissions `json:"rerun_permissions,omitempty"`
+	RerunPermissions RerunPermissions `json:"rerun_permissions,omitempty"`
 }
 
 type RerunPermissions struct {
 	// If AllowAnyone is set to true, any user can rerun the job
 	AllowAnyone bool `json:"allow_anyone,omitempty"`
 	// GitHubTeams contains IDs of GitHub teams of users who can rerun the job
+	// If you know the name of a team and the org it belongs to,
+	// you can look up its ID using this command, where the team slug is the hyphenated name:
+	// curl -H "Authorization: token <token>" "https://api.github.com/orgs/<org-name>/teams/<team slug>"
+	// or, to list all teams in a given org, use
+	// curl -H "Authorization: token <token>" "https://api.github.com/orgs/<org-name>/teams"
 	GitHubTeams []int `json:"github_teams,omitempty"`
```

Comment:
Can we make this match the slug rather than the ID?

Reply:
Yes, but the [GH api call](https://developer.github.com/v3/teams/members/#list-team-members) to list the team members requires the team ID, so we'd have to make double the requests: one to look up the ID using the slug, and one to look up the team members using the ID. I think it's worth the user inconvenience to require the ID here.

Reply:
The rest of prow and github disagree with you.

What is the team ID for the following two teams:
* https://github.com/kubernetes/org/blob/64811b88011a44321d002756fc63cf4c49b7d47c/config/kubernetes/sig-auth/teams.yaml#L13?
* https://github.com/istio/test-infra/blob/6b8b9a76a025487bb51a8daa8be133b62af6843b/prow/config.yaml#L73

Note that peribolos (something we wrote) and github branch protection (a github api) both use slugs, not ids.

It is orders of magnitude harder for a human to translate a slug into an ID than a machine. Although it is easier for the feature implementer to use IDs I think it is the wrong choice for the end user.

There's logic in here to consistently convert slugs into IDs: https://github.com/kubernetes/test-infra/tree/master/prow/cmd/peribolos

Reply:
I added a way to specify teams by slug and organization also, along with logic to look up the team ID by slug and org